python/lsst/ts/MTAlignment/mock/mock_t2sa.py

```diff
@@ -0,0 +1,926 @@
+                ),
+                rotation=BodyRotation(*np.random.normal(0.0, 6e-3, 3)),
+                radius=0.85,
+            ),
+        )
+
+        self._tel_az = 90.0
+        self._tel_alt = 0.0
+        self._tel_rot = 0.0
+
+        super().__init__(
+            name="MockT2SA",
+            host=host,
+            port=port,
+            connect_callback=self.run_reply_loop,
+            log=log,
+        )
+
+    @property
+    def reference_frame(self) -> str:
+        return f"FRAME{self._reference_group}"
+
+    def is_measuring(self) -> bool:
+        """Is t2sa executing a measurement?
+
+        Returns
+        -------
+        `bool`
+            True if there is an ongoing measurement.
+        """
+        return not self.measure_task.done()
+
+    def is_ready(self) -> bool:
+        """Determine is T2SA is ready for performing activities.
+
+        Returns
+        -------
+        `bool`
+            True if ready, False otherwise.
+        """
+        return self.laser_status == "LON" and self.laser_warmup_task.done()
+
+    def get_readiness_status(self) -> str:
+        """Get readiness status.
+
+        Returns
+        -------
+        `str`
+            A human readable string with the readiness status.
+        """
+        if self.laser_status != "LON":
+            return f"Laser status {self.laser_status}. Should be 'LON'."
+        else:
+            return "READY"
+
+    async def execute_drift(self, point_group: str) -> None:
+        """Pretend to execute two face check.
+
+        Parameters
+        ----------
+        point_group : `str`
+            Point group to execute drift measurements (e.g. M1M3, M2, CAM).
+        """
+        await self._execute_action("DRIFT", point_group)
+
+    async def execute_two_face_check(self, point_group: str) -> None:
+        """Pretend to execute two face check.
+
+        Parameters
+        ----------
+        point_group : `str`
+            Point group to execute 2 face check (e.g. M1M3, M2, CAM).
+        """
+        await self._execute_action("2FACE", point_group)
+
+    async def execute_measure_plan(self, point_group: str) -> None:
+        """Pretend to execute a measurement plan.
+
+        Acknowledge the request to measure, then pretend to measure.
+
+        Parameters
+        ----------
+        point_group : `str`
+            Point group to measure (e.g. M1M3, M2, CAM).
+        """
+        await self._execute_action("EMP", point_group)
+
+    async def halt(self) -> None:
+        """Halt any ongoing measurement."""
+        if not self.measure_task.done():
+            self.measure_task.cancel()
+            try:
+                await self.measure_task
+            except asyncio.CancelledError:
+                pass
+            except Exception:
+                self.log.exception("Error canceling measure task.")
+
+        self.t2sa_status = "READY"
+        await self.write_good_reply("READY")
```

Comment:
Please consider a named constant for "READY" to avoid errors caused by a typo, e.g. in future maintenance.